dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,441 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+            this.pageInDefaultLang = pageInDefaultLang;
+            this.pageInSecondLang = pageInSecondLang;
+        }
+    }
+
+    @DataProvider
+    public static Object[] siteSearchTestCases() {
+
+        /*
+         * Given sceneario: default-language content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in default language
+         * Expected: searching for the content in default language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case1 = new TestCaseSiteSearchBuilder()
+                .siteSearchDefaultLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+        /*
+         * Given sceneario: default-language content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in second language
+         * Expected: searching for the content in default language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case2 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+        /*
+         * Given sceneario: Content in second language only referenced from a page in default language.
+         * DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true.
+         * DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in second language.
+         * Expected: searching for content in second language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case3 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+        /*
+         * Same as above but with DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=false
+         * Expected: searching for content in second language should give NO results.
+         */
+
+        TestCaseSiteSearch case4 = new TestCaseSiteSearchBuilder()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'case4' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)